gists/bump-eks/README.md

```diff
@@ -0,0 +1,316 @@
+
+# Step 1: Download or update tools
+
+The upgrade script expects the following tools to exist on your machine, so make sure to install these.
+
+## aws CLI
+
+Follow instructions in https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
+
+## jq
+
+jq is a tool for parsing JSON.
+
+### Linux / apt
+
+```sh
+sudo apt-get install jq
+```
+
+### macOS / Linux via Homebrew
+
+```sh
+brew install jq
+```
+
+### Other
+
+See: https://stedolan.github.io/jq/download/.
+
+## yq
+
+yq is a tool for parsing YAML.
+
+### Linux / snap
+
+```sh
+snap install yq
+```
+
+###  macOS / Linux via Homebrew
+
+```sh
+brew install yq
+```
+
+### Other
+
+See https://github.com/mikefarah/yq.
+
+## watch
+
+### Linux
+
+No need to install, this ususally comes preinstalled in most distributions.
+
+### macOS
+
+```
+brew install watch
+```
+
+# Step 2: Log in to the environment
+
+Set `AWS_PROFILE` to the correct AWS profile from `~/.aws/config`. If you have not set this up, have a look at [Authenticating to AWS](https://www.okctl.io/authenticating-to-aws/#aws-single-sign-on-sso).
+
+```sh
+export AWS_PROFILE=my-dev-account
+aws sso login
+```
+
+# Step 3: Configure applications for no downtime
+
+
+
+To avoid downtime, **make sure** you have completed the steps described in this guide: https://github.com/oslokommune/okctl-upgrade/tree/main/gists/bump-eks-to-1-20#prepare-applications
+
+
+
+
+
+# Step 4: Adapt to EKS version specific requirements
+
+## EKS 1.21
+
+Make sure you have upgraded to Okctl version TODO or later.
+
+TODO: Okctl doesn't support EKS 1.21. We will update this guide when it is.
+
+## EKS 1.22
+
+If you're upgradring to EKS 1.22, you must follow the below steps. If you do not, your application **probably will stop working**.
+
+AWS describes the necessary changes we need to take into account in detail: https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html#update-1.22. However, we have attempted to extract everything that is necessary, so you shouldn't need to read that guide.
+
+### Update Ingress manifests
+
+First `cd` into the directory where you store your Kubernetes manifests/YAMLs. The default way in Okctl is to put these in your IAC repository.
+
+```yaml
+cd your-okctl-iac-repository
+```
+
+Find occurrences of old Ingress resources by running
+
+```
+grep -nRsH "apiVersion: networking.k8s.io/v1beta1"
+grep -nRsH "apiVersion: extensions/v1beta1"
+```
+
+Example output:
+
+```
+$ grep -nRsH "apiVersion: networking.k8s.io/v1beta1"
+--
+infrastructure/applications/okctl-reference-app/base/ingress.yaml:1:apiVersion: networking.k8s.io/v1beta1
+infrastructure/applications/okctl-reference-app/base/ingress.yaml-2-kind: Ingress
+--
+infrastructure/applications/hello/base/ingress.yaml:1:apiVersion: networking.k8s.io/v1beta1
+infrastructure/applications/hello/base/ingress.yaml-2-kind: Ingress
+```
+
+For every file in the result, edit it and replace the `apiVersion` so it becomes like this:
```

Comment:
Isn't there a change in how you describe routes between `extensions/v1beta1` and `networking.k8s.io/v1`

Reply:
True, good catch. I'll add it.